Makefile

```diff
@@ -13,4 +13,144 @@ collector: libkindling
+		InstallCommand = sudo yum -y install $(KernelName)
+	else
+		KernelName = NotSupport
+		CheckCommand = $(ErrorExit)
+		InstallCommand = $(ErrorExit)
+	endif
+endif
+
+LIBS_IMAGE ?= kindlingproject/kernel-builder:latest
+KINDLING_IMAGE ?= kindlingproject/agent-builder:latest
+
+
+## Check if probe build image and collector build image exists or pull from dockerhub
+.PHONY: builder-images
+builder-images:
+	@echo "Checking builder images...";
+	@if [ -z "$(shell docker images -q $(LIBS_IMAGE))" ]; then \
+		echo "Downloading probe build image..."; \
+		docker pull $(LIBS_IMAGE); \
+	else \
+		echo "Probe build image already exists"; \
+	fi
+	@if [ -z "$(shell docker images -q $(KINDLING_IMAGE))" ]; then \
+		echo "Downloading collector build image..."; \
+		docker pull $(KINDLING_IMAGE); \
+	else \
+		echo "Collector build image already exists"; \
+	fi
+
+
+## Check kernel headers exists or download, support Ubuntu, CentOS
+.PHONY: kernel-headers
+kernel-headers:
+	@echo "Checking kernel headers...";
+	@if [ "$(KernelName)" = "NotSupport" ]; then \
+		echo "System kernel is not the expected Ubuntu and CentOS.Try installing kernel headers locally"; \
+		$(ErrorExit); \
+	fi
+
+	@if [ -z "$(shell exit 1)" ]; then \
+		echo "Downloading $(KernelName)..."; \
+		$(InstallCommand); \
+	else \
+		echo "$(KernelName) already installed"; \
+	fi
+
+AGENT_LIBS_COMMIT_ID=$(shell cd $(LIBS_SRC) && git rev-parse HEAD)
+
+## Check libs exists or download, patch cmake file
+.PHONY: patch-libs
+patch-libs:
+	@echo "Checking libs...";
+	@if [ ! -d "$(LIBS_SRC)" ]; then \
+		echo "$(LIBS_SRC) not exist, downloading....."; \
+		git submodule update --init --recursive; \
+	else \
+		echo "Libs src: $(LIBS_SRC) exist"; \
+	fi
+	@echo "Patching libs commit id: $(AGENT_LIBS_COMMIT_ID)";
+
+	@echo "Patching $(PROBE_PATCH_FILE)";
+	@sed 's/AGENT_LIBS_VERSION "[0-9a-zA-Z]*"/AGENT_LIBS_VERSION "${AGENT_LIBS_COMMIT_ID}"/g' $(PROBE_PATCH_FILE) > $(PROBE_PATCH_FILE).new && \
+	mv $(PROBE_PATCH_FILE).new $(PROBE_PATCH_FILE);
+	@echo "Patching $(PROBE_PATCH_FILE) done"
+
+## Grant executable permissions to build.sh
+.PHONY: patch-scripts
+patch-scripts:
+	@echo "Grant executable permissions to build.sh";
+	@cd $(SCRIPTS_PATH) && chmod +x build.sh;
+
+## Check all depends
+.PHONY: depends
+depends: builder-images kernel-headers patch-libs patch-scripts
+
+## Build probe in docker with /lib/modules
+.PHONY: build-probe
+build-probe:
+	@echo "Building kindling probe..."
+	@docker run \
+		--env "ENV_HEADER_VERSION=$(kernelVersion)" \
+		--rm -it \
+		-v /usr:/host/usr \
+		-v /lib/modules:/host/lib/modules \
+		-v $(LIBS_SRC):/source \
+		$(LIBS_IMAGE);
+
+.PHONY: pack-probe
+pack-probe:
+	@echo "Packaging kindling probe...";
+	@if [ ! -d "$(LIBS_SRC)/kindling-falcolib-probe/" ]; then \
+		echo "The packaged probe does not exist.try again"; \
+		exit 1; \
+	else \
+		cd $(LIBS_SRC) && \
+		tar -cvzf kindling-falcolib-probe.tar.gz kindling-falcolib-probe/ && \
```

Comment:
Please add the `probe/libs/agent-libs/kindling-falcolib-probe.tar.gz` to the .gitignore of `agent-lib`, or just remove it after copy.

Reply:
Changed `cp -f` to `mv`

Reply:
Have also added `clean` to the makfile, clean the build file